src/lib.rs

```diff
@@ -43,59 +43,85 @@ use world::World;
 
 const TRACING_TARGET: &str = "turmoil";
 
+/// Utility method for performing a function on all hosts in `a` against all
+/// hosts in `b`.
+pub(crate) fn for_pairs(a: &Vec<IpAddr>, b: &Vec<IpAddr>, mut f: impl FnMut(IpAddr, IpAddr)) {
+    for first in a {
+        for second in b {
+            f(*first, *second)
+        }
+    }
+}
+
 /// Lookup an ip address by host name.
 ///
 /// Must be called from within a Turmoil simulation.
 pub fn lookup(addr: impl ToIpAddr) -> IpAddr {
     World::current(|world| world.lookup(addr))
 }
 
-/// Hold messages two hosts, until [`release`] is called.
+/// Lookup an IP address by host name. Use regex to match a number of hosts.
+///
+/// Must be called from within a Turmoil simulation.
+pub fn lookup_many(addr: impl ToIpAddrs) -> Vec<IpAddr> {
```

Comment:
Seems only a Regex would be useful here, so perhaps we can just be explicit instead of using the trait?

Reply:
The `ToIpAddrs` trait could also support a Vec of string matches (actually not with this revision, but I'll update it), so for example another valid input would be:

```rust
lookup_many(vec!["host-a", "host-b"]);
```

I was also looking into passing predicate callbacks, but I ran into trouble with conflicting generics with the `ToIpAddr` impl for `ToIpAddrs`.

Changing this, however, will affect all the other methods that currently take `ToIpAddr` (hold, release, partition). It may be a nice convenience if we asserted that the `lookup_many` should produce at least one element in each set, as its use could otherwise produce subtle bugs.

Reply:
> It may be a nice convenience if we asserted that the lookup_many should produce at least one element in each set, as its use could otherwise produce subtle bugs.

Does this mean that if we look up two hosts we must see at least two hosts returning?